pallets/protos/src/lib.rs

```diff
@@ -796,10 +798,14 @@ pub mod pallet {
 				},
 			};
 
-			ensure!(!<DetachedHashes<T>>::contains_key(&proto_hash), Error::<T>::Detached);
+			let detach_hash = DetachHash::Proto(proto_hash);
+			let detach_request = DetachRequest { hash: detach_hash.clone(), target_chain, target_account };
+
+			ensure!(!<DetachedHashes<T>>::contains_key(&detach_hash), Error::<T>::Detached);
+			ensure!(!<DetachRequests<T>>::get().contains(&detach_request), Error::<T>::DetachRequestAlreadyExists);
```

Comment:
@sinkingsugar is there any need to check to check if the detach requests already exists. I don't think there is, but I just added this line to be overly cautious...because why not 😅